<?xml version="1.0"?>

<launch>

  <!-- "Disable" odometry from azimut3 -->
  <group ns="base_controller">
      <param name="odom_frame_id" type="string" value="az3_odom"/>
      <param name="base_frame_id" type="string" value="az3_base_link"/>
  </group>
  <remap from="/base_controller/odom" to="/base_controller/az3_odom"/>

  <!-- use same launch file as "Kinect + Odometry" configuration -->
  <include file="$(find rtabmap_legacy)/launch/azimut3/az3_mapping_robot_kinect_odom.launch"/>

  <!-- Odometry -->
  <node pkg="rtabmap_odom" type="visual_odometry" name="visual_odometry" output="screen">
    <remap from="rgb/image"       to="/camera/rgb/image_rect_color"/>
    <remap from="depth/image"     to="/camera/depth_registered/image_raw"/>
    <remap from="rgb/camera_info" to="/camera/depth_registered/camera_info"/>
    <remap from="odom" to="/base_controller/odom"/>

    <param name="Odom/MinInliers" type="string" value="10"/>
    <param name="Odom/InlierDistance" type="string" value="0.01"/>

    <param name="frame_id" type="string" value="base_footprint"/>
  </node>

</launch>
